### ISSUE TITLE
[Ring] RequestError: getaddrinfo ENOTFOUND api.ring.com

### ISSUE THREAD

**Issue by aterlyga** [dgreif/ring#840]
# Bug Report

### Describe the Bug
The plugin installs, configuration goes well but then it is not functional (video not showing, no reaction for doorbell button) and there is error message in log file every 5 seconds since the plugin starts. 

### To Reproduce
 - Start Homebridge
 - Attempt to open a video from doorbell in home app
 - Press the doorbell button on Ring doorbell
 - The video would be expected to be shown, but it's not
 - There is no reaction, neither there are log records about the button pressing

### Expected behavior
The video would be shown
The HomePod gives a chime + there is a log record about the button press in the log

### Screenshots/Logs
[12/24/2021, 9:24:41 PM] [Ring] Failed to reach Ring server at https://api.ring.com/clients_api/dings/active.  getaddrinfo ENOTFOUND api.ring.com.  Trying again in 5 seconds...
[12/24/2021, 9:24:41 PM] [Ring] RequestError: getaddrinfo ENOTFOUND api.ring.com
    at ClientRequest.<anonymous> (/homebridge/node_modules/homebridge-ring/node_modules/got/dist/source/core/index.js:956:111)
    at Object.onceWrapper (events.js:422:26)
    at ClientRequest.emit (events.js:327:22)
    at ClientRequest.origin.emit (/homebridge/node_modules/homebridge-ring/node_modules/@szmarczak/http-timer/dist/source/index.js:43:20)
    at TLSSocket.socketErrorListener (_http_client.js:469:9)
    at TLSSocket.emit (events.js:315:20)
    at emitErrorNT (internal/streams/destroy.js:106:8)
    at emitErrorCloseNT (internal/streams/destroy.js:74:3)
    at processTicksAndRejections (internal/process/task_queues.js:80:21)
    at GetAddrInfoReqWrap.onlookup [as oncomplete] (dns.js:67:26) {
  code: 'ENOTFOUND',
  timings: {
    start: 1640373881091,
    socket: 1640373881091,
    lookup: 1640373881138,
    connect: undefined,
    secureConnect: undefined,
    upload: undefined,
    response: undefined,
    end: undefined,
    error: 1640373881138,
    abort: undefined,
    phases: {
      wait: 0,
      dns: 47,
      tcp: undefined,
      tls: undefined,
      request: undefined,
      firstByte: undefined,
      download: undefined,
      total: 47
    }
  }
}
[12/24/2021, 9:24:44 PM] [Ring] Failed to reach Ring server at https://api.ring.com/clients_api/ring_devices.  getaddrinfo ENOTFOUND api.ring.com.  Trying again in 5 seconds...
[12/24/2021, 9:24:44 PM] [Ring] RequestError: getaddrinfo ENOTFOUND api.ring.com
    at ClientRequest.<anonymous> (/homebridge/node_modules/homebridge-ring/node_modules/got/dist/source/core/index.js:956:111)
    at Object.onceWrapper (events.js:422:26)
    at ClientRequest.emit (events.js:327:22)
    at ClientRequest.origin.emit (/homebridge/node_modules/homebridge-ring/node_modules/@szmarczak/http-timer/dist/source/index.js:43:20)
    at TLSSocket.socketErrorListener (_http_client.js:469:9)
    at TLSSocket.emit (events.js:315:20)
    at emitErrorNT (internal/streams/destroy.js:106:8)
    at emitErrorCloseNT (internal/streams/destroy.js:74:3)
    at processTicksAndRejections (internal/process/task_queues.js:80:21)
    at GetAddrInfoReqWrap.onlookup [as oncomplete] (dns.js:67:26) {
  code: 'ENOTFOUND',
  timings: {
    start: 1640373884516,
    socket: 1640373884516,
    lookup: 1640373884563,
    connect: undefined,
    secureConnect: undefined,
    upload: undefined,
    response: undefined,
    end: undefined,
    error: 1640373884563,
    abort: undefined,
    phases: {
      wait: 0,
      dns: 47,
      tcp: undefined,
      tls: undefined,
      request: undefined,
      firstByte: undefined,
      download: undefined,
      total: 47
    }
  }
}

### Additional context
None

### Homebridge Ring Config
Post homebridge-ring platform config  _without sensitive information_
```

{
    "refreshToken": "[token removed as sensitive info]",
    "debug": true,
    "platform": "Ring"
}

```

### Environment

 - OS: Alpine Linux (3.12.1)
 - Node.js: v14.15.1
 - NPM: v6.14.8
 - homebridge-ring: 9.23.0
 - homebridge: v1.3.8
 - hoobs: N/A

**Comment by aterlyga** [comment]
Small update. I've managed to have the video and button working one time, right after Homebridge reboot. But in a while the issue started to appear again. Here is a log file since the start and until the issue starts to take place (only Ring logs kept).

[12/24/2021, 10:38:56 PM] [HB Supervisor] Restarting Homebridge...
[12/24/2021, 10:38:57 PM] [HB Supervisor] Starting Homebridge with extra flags: -I -P /homebridge/node_modules
[12/24/2021, 10:38:59 PM] [HB Supervisor] Started Homebridge v1.3.8 with PID: 3066
[12/24/2021, 10:39:02 PM] Loaded config.json with 0 accessories and 5 platforms.
[12/24/2021, 10:39:03 PM] Loaded 117 cached accessories from cachedAccessories.
[12/24/2021, 10:39:08 PM] ---
[12/24/2021, 10:39:11 PM] Loaded plugin: homebridge-ring@9.23.0
[12/24/2021, 10:39:11 PM] Registering platform 'homebridge-ring.Ring'
[12/24/2021, 10:39:14 PM] ---
[12/24/2021, 10:39:14 PM] Loading 5 platforms...
[12/24/2021, 10:39:14 PM] [Config] Initializing config platform...
[12/24/2021, 10:39:14 PM] [Config] Running in Service Mode
[12/24/2021, 10:39:14 PM] [Ring] Initializing Ring platform...
[12/24/2021, 10:39:24 PM] [Ring] Configuring cached accessory 50e83370-a338-4500-8ed2-8298c137c17d Front Door
[12/24/2021, 10:39:24 PM] [Ring] Configuring cached accessory 1e70a9a2-8cee-4866-9520-a87fddef2af5 Street Door
[12/24/2021, 10:39:35 PM] Homebridge v1.3.8 (HAP v0.9.7) (Homebridge FDED) is running on port 51737.
[12/24/2021, 10:39:36 PM] [Ring] Found the following locations:
[12/24/2021, 10:39:36 PM] [Ring]   locationId: 8d58b1e1-5d3f-4ccc-a99f-4fc7edc73f2b - Sokilnyky
[12/24/2021, 10:39:37 PM] [Ring] Location Mode: {"mode":"unset","lastUpdateTimeMS":1628269947795,"securityStatus":{},"readOnly":true,"notYetParticipatingInMode":[],"responseTimestamp":1640378377000}
[12/24/2021, 10:39:37 PM] [Ring] Configuring 2 cameras and 2 devices for location "XX" - locationId: YY
[12/24/2021, 10:39:59 PM] [Ring] Preparing Live Stream for Front Door
[12/24/2021, 10:40:00 PM] [Ring] Connecting to Front Door using STUN
[12/24/2021, 10:40:00 PM] [Ring] Waiting for stream data from Front Door (1.308s)
[12/24/2021, 10:40:02 PM] [Ring] Video stream latched for Front Door
[12/24/2021, 10:40:02 PM] [Ring] Received stream data from Front Door (2.571s)
[12/24/2021, 10:40:02 PM] [Ring] Streaming active for Front Door
[12/24/2021, 10:40:02 PM] [Ring] Audio stream latched for Front Door
[12/24/2021, 10:40:02 PM] [Ring] From Ring (Front Door): Guessed Channel Layout for Input Stream #0.0 : mono
Input #0, sdp, from 'pipe:':
  Metadata:
    title           : Talk
  Duration: N/A, start: 0.000000, bitrate: 64 kb/s
    Stream #0:0: Audio: pcm_mulaw, 8000 Hz, mono, s16, 64 kb/s
Stream mapping:
  Stream #0:0 -> #0:0 (pcm_mulaw (native) -> aac (libfdk_aac))
Output #0, rtp, to 'srtp://192.168.88.2:57903?localrtcpport=11263&pkt_size=188':
  Metadata:
    title           : Talk
    encoder         : Lavf58.45.100
    Stream #0:0: Audio: aac (libfdk_aac) (ELD), 16000 Hz, mono, s16, 24 kb/s
    Metadata:
      encoder         : Lavc58.91.100 libfdk_aac

[12/24/2021, 10:40:02 PM] [Ring] From Ring (Front Door): size=       3kB time=00:00:00.81 bitrate=  30.4kbits/s speed=1.44x    
[12/24/2021, 10:40:03 PM] [Ring] From Ring (Front Door): size=       5kB time=00:00:01.35 bitrate=  29.4kbits/s speed=1.26x    
[12/24/2021, 10:40:03 PM] [Ring] From Ring (Front Door): size=       7kB time=00:00:01.87 bitrate=  29.0kbits/s speed=1.15x    
[12/24/2021, 10:40:04 PM] [Ring] requesting key frame
[12/24/2021, 10:40:04 PM] [Ring] From Ring (Front Door): size=       9kB time=00:00:02.47 bitrate=  28.8kbits/s speed=1.13x    
[12/24/2021, 10:40:05 PM] [Ring] From Ring (Front Door): size=      11kB time=00:00:03.02 bitrate=  28.6kbits/s speed= 1.1x    
[12/24/2021, 10:40:05 PM] [Ring] From Ring (Front Door): size=      12kB time=00:00:03.56 bitrate=  28.5kbits/s speed=1.09x    
[12/24/2021, 10:40:06 PM] [Ring] From Ring (Front Door): size=      14kB time=00:00:04.11 bitrate=  28.5kbits/s speed=1.05x    
[12/24/2021, 10:40:06 PM] [Ring] From Ring (Front Door): size=      16kB time=00:00:04.71 bitrate=  28.4kbits/s speed=1.06x    
[12/24/2021, 10:40:07 PM] [Ring] From Ring (Front Door): size=      18kB time=00:00:05.23 bitrate=  28.4kbits/s speed=1.06x    
[12/24/2021, 10:40:07 PM] [Ring] From Ring (Front Door): size=      20kB time=00:00:05.77 bitrate=  28.4kbits/s speed=1.05x    
[12/24/2021, 10:40:08 PM] [Ring] From Ring (Front Door): size=      21kB time=00:00:06.15 bitrate=  28.3kbits/s speed=1.03x    
[12/24/2021, 10:40:09 PM] [Ring] Front Door Doorbell Pressed
[12/24/2021, 10:40:09 PM] [Ring] From Ring (Front Door): size=      23kB time=00:00:06.63 bitrate=  28.3kbits/s speed=0.967x    
[12/24/2021, 10:40:09 PM] [Ring] From Ring (Front Door): size=      25kB time=00:00:07.71 bitrate=  26.7kbits/s speed=1.05x    
[12/24/2021, 10:40:10 PM] [Ring] From Ring (Front Door): size=      27kB time=00:00:08.19 bitrate=  26.8kbits/s speed=1.04x    
[12/24/2021, 10:40:10 PM] [Ring] From Ring (Front Door): size=      28kB time=00:00:08.67 bitrate=  26.8kbits/s speed=1.03x    
[12/24/2021, 10:40:11 PM] [Ring] Stopped Live Stream for Front Door
[12/24/2021, 10:40:11 PM] [Ring] Return Audio (Front Door): stopped gracefully
[12/24/2021, 10:40:11 PM] [Ring] Return Audio (Front Door): Guessed Channel Layout for Input Stream #0.0 : mono
Input #0, sdp, from 'pipe:':
  Metadata:
    title           : Talk
  Duration: N/A, bitrate: N/A
    Stream #0:0: Audio: aac, 16000 Hz, mono, s16
Stream mapping:
  Stream #0:0 -> #0:0 (aac (libfdk_aac) -> pcm_mulaw (native))
Finishing stream 0:0 without any data written to it.
Output #0, rtp, to 'srtp://127.0.0.1:48833?pkt_size=188':
  Metadata:
    title           : Talk
    encoder         : Lavf58.45.100
    Stream #0:0: Audio: pcm_mulaw, 8000 Hz, mono, s16, 64 kb/s
    Metadata:
      encoder         : Lavc58.91.100 pcm_mulaw
size=       0kB time=00:00:00.00 bitrate=N/A speed=   0x    
video:0kB audio:0kB subtitle:0kB other streams:0kB global headers:0kB muxing overhead: unknown
Output file is empty, nothing was encoded (check -ss / -t / -frames parameters if used)
Exiting normally, received signal 15.

[12/24/2021, 10:40:17 PM] [Ring] Preparing Live Stream for Front Door
[12/24/2021, 10:40:17 PM] [Ring] Ding 7045371581123444349 is expired (480).  Fetching a new ding and trying video stream again
err:  Error: remote peer disconnected
    at TLSSocket.<anonymous> (/homebridge/node_modules/homebridge-ring/lib/api/sip.js:557:38)
    at TLSSocket.emit (events.js:327:22)
    at endReadableNT (_stream_readable.js:1327:12)
    at processTicksAndRejections (internal/process/task_queues.js:80:21)
[12/24/2021, 10:40:18 PM] [Ring] Connecting to Front Door using STUN
[12/24/2021, 10:40:18 PM] [Ring] Waiting for stream data from Front Door (1.368s)
[12/24/2021, 10:40:20 PM] [Ring] Video stream latched for Front Door
[12/24/2021, 10:40:20 PM] [Ring] Received stream data from Front Door (3.2s)
[12/24/2021, 10:40:20 PM] [Ring] Stopped Live Stream for Front Door
[12/24/2021, 10:40:20 PM] [Ring] From Ring (Front Door): stopped gracefully
[12/24/2021, 10:40:20 PM] [Ring] From Ring (Front Door): Guessed Channel Layout for Input Stream #0.0 : mono
Input #0, sdp, from 'pipe:':
  Metadata:
    title           : Talk
  Duration: N/A, bitrate: 64 kb/s
    Stream #0:0: Audio: pcm_mulaw, 8000 Hz, mono, s16, 64 kb/s
Stream mapping:
  Stream #0:0 -> #0:0 (pcm_mulaw (native) -> aac (libfdk_aac))
Finishing stream 0:0 without any data written to it.
Output #0, rtp, to 'srtp://192.168.88.2:53847?localrtcpport=17551&pkt_size=188':
  Metadata:
    title           : Talk
    encoder         : Lavf58.45.100
    Stream #0:0: Audio: aac (libfdk_aac) (ELD), 16000 Hz, mono, s16, 24 kb/s
    Metadata:
      encoder         : Lavc58.91.100 libfdk_aac
[libfdk_aac @ 0x7fa5d0f01b40] Trying to remove 512 samples, but the queue is empty
size=       0kB time=00:00:00.00 bitrate=N/A speed=   0x    
video:0kB audio:0kB subtitle:0kB other streams:0kB global headers:0kB muxing overhead: unknown
Exiting normally, received signal 15.

[12/24/2021, 10:40:20 PM] [Ring] Return Audio (Front Door): stopped gracefully
[12/24/2021, 10:40:20 PM] [Ring] Return Audio (Front Door): Guessed Channel Layout for Input Stream #0.0 : mono
Input #0, sdp, from 'pipe:':
  Metadata:
    title           : Talk
  Duration: N/A, bitrate: N/A
    Stream #0:0: Audio: aac, 16000 Hz, mono, s16
Stream mapping:
  Stream #0:0 -> #0:0 (aac (libfdk_aac) -> pcm_mulaw (native))
Finishing stream 0:0 without any data written to it.

[12/24/2021, 10:40:20 PM] [Ring] Return Audio (Front Door): Output #0, rtp, to 'srtp://127.0.0.1:52043?pkt_size=188':
  Metadata:
    title           : Talk
    encoder         : Lavf58.45.100
    Stream #0:0: Audio: pcm_mulaw, 8000 Hz, mono, s16, 64 kb/s
    Metadata:
      encoder         : Lavc58.91.100 pcm_mulaw
size=       0kB time=00:00:00.00 bitrate=N/A speed=   0x    
video:0kB audio:0kB subtitle:0kB other streams:0kB global headers:0kB muxing overhead: unknown
Output file is empty, nothing was encoded (check -ss / -t / -frames parameters if used)
Exiting normally, received signal 15.

[12/24/2021, 10:40:21 PM] [Ring] From Ring (Front Door): stopped gracefully
[12/24/2021, 10:40:21 PM] [Ring] From Ring (Front Door): pipe:: Operation timed out
size=      30kB time=00:00:09.03 bitrate=  26.9kbits/s speed=0.482x    
video:0kB audio:25kB subtitle:0kB other streams:0kB global headers:0kB muxing overhead: 16.897720%

[12/24/2021, 10:40:21 PM] [Ring] From Ring (Front Door): Exiting normally, received signal 15.


[12/24/2021, 10:40:24 PM] [Ring] Failed to reach Ring server at https://api.ring.com/clients_api/dings/active.  getaddrinfo ENOTFOUND api.ring.com.  Trying again in 5 seconds...
[12/24/2021, 10:40:24 PM] [Ring] RequestError: getaddrinfo ENOTFOUND api.ring.com
    at ClientRequest.<anonymous> (/homebridge/node_modules/homebridge-ring/node_modules/got/dist/source/core/index.js:956:111)
    at Object.onceWrapper (events.js:422:26)
    at ClientRequest.emit (events.js:327:22)
    at ClientRequest.origin.emit (/homebridge/node_modules/homebridge-ring/node_modules/@szmarczak/http-timer/dist/source/index.js:43:20)
    at TLSSocket.socketErrorListener (_http_client.js:469:9)
    at TLSSocket.emit (events.js:315:20)
    at emitErrorNT (internal/streams/destroy.js:106:8)
    at emitErrorCloseNT (internal/streams/destroy.js:74:3)
    at processTicksAndRejections (internal/process/task_queues.js:80:21)
    at GetAddrInfoReqWrap.onlookup [as oncomplete] (dns.js:67:26) {
  code: 'ENOTFOUND',
  timings: {
    start: 1640378423907,
    socket: 1640378423907,
    lookup: 1640378423953,
    connect: undefined,
    secureConnect: undefined,
    upload: undefined,
    response: undefined,
    end: undefined,
    error: 1640378423955,
    abort: undefined,
    phases: {
      wait: 0,
      dns: 46,
      tcp: undefined,
      tls: undefined,
      request: undefined,
      firstByte: undefined,
      download: undefined,
      total: 48
    }
  }
}
[12/24/2021, 10:40:29 PM] [Ring] Failed to reach Ring server at https://api.ring.com/clients_api/dings/active.  getaddrinfo ENOTFOUND api.ring.com.  Trying again in 5 seconds...
[12/24/2021, 10:40:29 PM] [Ring] RequestError: getaddrinfo ENOTFOUND api.ring.com
    at ClientRequest.<anonymous> (/homebridge/node_modules/homebridge-ring/node_modules/got/dist/source/core/index.js:956:111)
    at Object.onceWrapper (events.js:422:26)
    at ClientRequest.emit (events.js:327:22)
    at ClientRequest.origin.emit (/homebridge/node_modules/homebridge-ring/node_modules/@szmarczak/http-timer/dist/source/index.js:43:20)
    at TLSSocket.socketErrorListener (_http_client.js:469:9)
    at TLSSocket.emit (events.js:315:20)
    at emitErrorNT (internal/streams/destroy.js:106:8)
    at emitErrorCloseNT (internal/streams/destroy.js:74:3)
    at processTicksAndRejections (internal/process/task_queues.js:80:21)
    at GetAddrInfoReqWrap.onlookup [as oncomplete] (dns.js:67:26) {
  code: 'ENOTFOUND',
  timings: {
    start: 1640378429190,
    socket: 1640378429191,
    lookup: 1640378429237,
    connect: undefined,
    secureConnect: undefined,
    upload: undefined,
    response: undefined,
    end: undefined,
    error: 1640378429237,
    abort: undefined,
    phases: {
      wait: 1,
      dns: 46,
      tcp: undefined,
      tls: undefined,
      request: undefined,
      firstByte: undefined,
      download: undefined,
      total: 47
    }
  }
}
[12/24/2021, 10:40:34 PM] [Ring] Failed to reach Ring server at https://api.ring.com/clients_api/dings/active.  getaddrinfo ENOTFOUND api.ring.com.  Trying again in 5 seconds...
[12/24/2021, 10:40:34 PM] [Ring] RequestError: getaddrinfo ENOTFOUND api.ring.com
    at ClientRequest.<anonymous> (/homebridge/node_modules/homebridge-ring/node_modules/got/dist/source/core/index.js:956:111)
    at Object.onceWrapper (events.js:422:26)
    at ClientRequest.emit (events.js:327:22)
    at ClientRequest.origin.emit (/homebridge/node_modules/homebridge-ring/node_modules/@szmarczak/http-timer/dist/source/index.js:43:20)
    at TLSSocket.socketErrorListener (_http_client.js:469:9)
    at TLSSocket.emit (events.js:315:20)
    at emitErrorNT (internal/streams/destroy.js:106:8)
    at emitErrorCloseNT (internal/streams/destroy.js:74:3)
    at processTicksAndRejections (internal/process/task_queues.js:80:21)
    at GetAddrInfoReqWrap.onlookup [as oncomplete] (dns.js:67:26) {
  code: 'ENOTFOUND',
  timings: {
    start: 1640378434244,
    socket: 1640378434244,
    lookup: 1640378434292,
    connect: undefined,
    secureConnect: undefined,
    upload: undefined,
    response: undefined,
    end: undefined,
    error: 1640378434292,
    abort: undefined,
    phases: {
      wait: 0,
      dns: 48,
      tcp: undefined,
      tls: undefined,
      request: undefined,
      firstByte: undefined,
      download: undefined,
      total: 48
    }
  }
}
[12/24/2021, 10:40:37 PM] [Ring] Failed to reach Ring server at https://api.ring.com/clients_api/ring_devices.  getaddrinfo ENOTFOUND api.ring.com.  Trying again in 5 seconds...
[12/24/2021, 10:40:37 PM] [Ring] RequestError: getaddrinfo ENOTFOUND api.ring.com
    at ClientRequest.<anonymous> (/homebridge/node_modules/homebridge-ring/node_modules/got/dist/source/core/index.js:956:111)
    at Object.onceWrapper (events.js:422:26)
    at ClientRequest.emit (events.js:327:22)
    at ClientRequest.origin.emit (/homebridge/node_modules/homebridge-ring/node_modules/@szmarczak/http-timer/dist/source/index.js:43:20)
    at TLSSocket.socketErrorListener (_http_client.js:469:9)
    at TLSSocket.emit (events.js:315:20)
    at emitErrorNT (internal/streams/destroy.js:106:8)
    at emitErrorCloseNT (internal/streams/destroy.js:74:3)
    at processTicksAndRejections (internal/process/task_queues.js:80:21)
    at GetAddrInfoReqWrap.onlookup [as oncomplete] (dns.js:67:26) {
  code: 'ENOTFOUND',
  timings: {
    start: 1640378437320,
    socket: 1640378437321,
    lookup: 1640378437363,
    connect: undefined,
    secureConnect: undefined,
    upload: undefined,
    response: undefined,
    end: undefined,
    error: 1640378437364,
    abort: undefined,
    phases: {
      wait: 1,
      dns: 42,
      tcp: undefined,
      tls: undefined,
      request: undefined,
      firstByte: undefined,
      download: undefined,
      total: 44
    }
  }
}

**Comment by dgreif** [comment]
Can you try updating to the latest version of Node.js (16.13.1 currently) and see if the issue still happens?

**Comment by aterlyga** [comment]
Just upgraded to 16.13.1, but unfortunately the issue persists

**Comment by dgreif** [comment]
Thanks for giving that a try.  This issue appears to be DNS related.  Can you try running `dig api.ring.com` from a terminal on your homebridge server and post the results here?

**Comment by aterlyga** [comment]
Thanks for your reply!
Looks like "dig" command is perhaps not supported in Docker on Synology or am I doing something wrong?
![image](https://user-images.githubusercontent.com/535428/147503909-af87778a-ae1b-481a-85e1-0269d275b03c.png)

**Comment by dgreif** [comment]
Do you by chance use the homebridge-notification plugin?

**Comment by x5nder** [comment]
Looks like I have the same issue here (last part of the location string censored);

```
[12/28/2021, 12:25:52 PM] [Ring] Failed to reach Ring server at https://app.ring.com/api/v1/mode/location/4e765a56-d85e-4f6c-5bd6-xxxxxxxxxxxx.  getaddrinfo EAI_AGAIN app.ring.com.  Trying again in 5 seconds...
[12/28/2021, 12:26:47 PM] [Ring] Failed to reach Ring server at https://api.ring.com/clients_api/dings/active.  getaddrinfo EAI_AGAIN api.ring.com.  Trying again in 5 seconds...
```

Dig results:

```
; <<>> DiG 9.16.8-Ubuntu <<>> api.ring.com
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 38439
;; flags: qr rd ra; QUERY: 1, ANSWER: 9, AUTHORITY: 0, ADDITIONAL: 1

;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags:; udp: 65494
;; QUESTION SECTION:
;api.ring.com.			IN	A

;; ANSWER SECTION:
api.ring.com.		0	IN	CNAME	clientsapigw.ring.com.
clientsapigw.ring.com.	31	IN	A	54.87.5.25
clientsapigw.ring.com.	31	IN	A	52.22.96.159
clientsapigw.ring.com.	31	IN	A	52.55.223.197
clientsapigw.ring.com.	31	IN	A	184.72.251.35
clientsapigw.ring.com.	31	IN	A	34.233.53.185
clientsapigw.ring.com.	31	IN	A	52.5.95.72
clientsapigw.ring.com.	31	IN	A	75.101.151.51
clientsapigw.ring.com.	31	IN	A	54.158.73.119

;; Query time: 0 msec
;; SERVER: 127.0.0.53#53(127.0.0.53)
;; WHEN: di dec 28 12:34:53 CET 2021
;; MSG SIZE  rcvd: 196
```

**Comment by dgreif** [comment]
Do either of you have an Eero router by chance?  If so, check out https://github.com/homebridge/homebridge-raspbian-image/wiki/How-To-Fix-Eero-Router-Internet-and-DNS-Connectivity-Issues

**Comment by x5nder** [comment]
I have a mesh wifi system from TP-Link _but_ my Homebridge server is hardwired to my network through ethernet...

**Comment by aterlyga** [comment]
I am sorry for delay with my replies - I was out during holidays season.

> Do you by chance use the homebridge-notification plugin?

I don't. I have 4 plugins:
- Homebridge UI
- Eveatmo
- FibaroHC2
- Ring

> Do either of you have an Eero router by chance? If so, check out https://github.com/homebridge/homebridge-raspbian-image/wiki/How-To-Fix-Eero-Router-Internet-and-DNS-Connectivity-Issues

No, I am using Mikrotik router.

**Comment by tsightler** [comment]
I'd still be very suspicious this is a conflict with another plugin using outdated dependencies that are patching the underlying http functions. The Eveatmo plugin itself doesn't look too bad at first, but digging a little deeper it uses an older API that hasn't had an update in 3+ years and that in turn uses some pretty outdated libraries that are known to cause problems with got, which is the HTTP library used by the Ring plugin.

I'd suggest trying to reproduce the issue with just the Ring plugin to verify it's not a conflict.

**Comment by aterlyga** [comment]
Thanks so much for your reply. I will give it a try with just Ring plugin in the evening. Will keep you updated!

**Comment by aterlyga** [comment]
I tried without Eveatmo and unfortunately the issue persists. I was trying to somehow also run without FibaroHC2, but here I struggle to do it without putting down my large Fibaro-based ecosystem.
However, I tried something else in Homebridge terminal:
![image](https://user-images.githubusercontent.com/535428/150194205-640dea62-acc7-4f06-a863-95842c61ff66.png)
Looks like ring.com resolves, google.com resolves, but api.ring.com does not.
Does it mean that the issue is not in plugin(s) code, but is somewhere on my Synology DSM? If I try to ping api.ring.com on my laptop in the same network - it resolves..
